specs/0108-community-health-dashboard.md

```diff
@@ -0,0 +1,49 @@
+- Feature Name: Community Health Dashboard
+- Created: 2020-10-23
+- Associated Bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1672966
+
+# Summary
+
+Create a team-specific health dashboard targeted at Pontoon users with Manager rights.
+
+# Motivation
+
+Team managers as well as project managers should have the ability to understand how localization communities are performing. Team dashboards already provide some of the answers, including the overall status of completion and translation statistics, but the information is only available in the form of a current snapshot.
+
+There's no historic data to provide insights into how the community health is developing. Missing is the information on how completion and volume of unreviewed strings are changing over time. It's not clear what share of translations was copied from the Machinery. We also don't know the ratio of translations approved without a peer review.
+
+# Out of scope
+
+At this stage we’re only interested in presenting data we already collect in Pontoon. We're primarily focusing on exposing data to team managers; project managers will be able to obtain data across all teams if needed, e.g. by using scripts and the API.
+
+# Feature explanation
+
+A new "Insights" tab is made available on the Team page, positioned next to the "Contributors" tab. It consists of several sections, presenting data for the period of the last 12 months. Each section is described below. Fore more details about the design, see the [Mockup](#mockup) section.
+
+## Active users
+
+Pie charts showing ratios of active vs. all managers, reviewers and contributors. See the existing scripts ([1](https://github.com/flodolo/scripts/blob/954fa85/pontoon/active_contributors.py), [2](https://github.com/flodolo/scripts/blob/954fa85/pontoon/list_reviewers_with_contribution_stats.py)) for criteria of defining active users and for details on how to retrieve data.
+
+## Unreviewed suggestion lifespan
+
+A line chart showing time before suggestion gets approved or rejected. See the [existing script](https://github.com/flodolo/scripts/blob/954fa85/pontoon/unreviewed_suggestions_lifespan.py) for details on how to retrieve data. A tooltip showing the exact data at a given time appears when hovering over a chart.
+
+## Translation activity
```

Comment:
Ok, looking at the dashboard, why are we including a bar for new source strings added? New source strings come in independent of localizer engagement and don't really serve as an interesting point of reference within this context (especially when the translation activity stats are significantly higher volume than the new source strings). Could we make human v machinery into two columns and remove the new source strings column? Also, instead of raw volume, could the tooltip be a percentage of the total, so we can see the ratio of human translations vs machinery translations more clearly?

Reply:
IMO the new string bar is useful to understand what happened: did they translate more because there was more to translate, or are they doing QA?

Reply:
We have other, more concrete, ways of depicting what happened though. For instance, each string has a translation history. If that history log increases, then we can be sure its QA. If history is created for the first time, we can be sure that it's a new translation. With the bar, we're left to make assumptions rather than knowing for certain.

Reply:
True that, although I don't know if that information can be collected easily.

Having said that, knowing how many strings are "requested" every month is something I'd like to know, so maybe it should be in its own graph?

Reply:
I don't know either, and maybe that's something to include in the out-of-scope list (Matjaz can make that call).

Agreed, that is an interesting metric that might be better suited to its own graph.